subsys/pm/device.c

```diff
@@ -25,10 +22,14 @@ static size_t num_susp;
 
 static int _pm_devices(enum pm_device_state state)
 {
-	const struct device *dev;
+	const struct device *devs;
+	size_t devc;
+
+	devc = z_device_get_all_static(&devs);
+
 	num_susp = 0;
 
-	for (dev = (__device_end - 1); dev > __device_start; dev--) {
+	for (const struct device *dev = devs + devc - 1; dev > devs; dev--) {
```

Comment:
Isn't this off-by-one (before and after this commit)?

It looks like it should be `dev >= devs`, since `devs` points at the first valid device, right?

Reply:
Although I guess you can only point one past the end of an array and not necessarily one before the beginning...

Reply:
You're right, `dev >= devs` allows reaching up to the first entry. Looks like we always had this problem... fixed